### PR TITLE
Lock down systemd service

### PR DESCRIPTION
**Pull request by ecksun**
To be on the safe side, restrict what the service can do.

### CODE REVIEW

**Comment by ecksun** [comment]
I added `ProtectControlGroups` and `ProtectKernelModules` and changed `ProtectSystem=strict`.

Do you want me to do the check for an old systemd version also?

**Comment by ecksun** [comment]
Alright, the other Protect directives also require 232 according to [the changelog](https://github.com/systemd/systemd/blob/master/NEWS) so I included them in the version check as well

File: tasks/preflight.yml [hunk 1]
```diff
@@ -15,6 +15,12 @@
   when: item in node_exporter_enabled_collectors
   with_items: "{{ node_exporter_disabled_collectors }}"
 
+- name: Get systemd version
+  shell: systemctl --version | awk '$1 == "systemd" {print $2}'
+  changed_when: false
+  check_mode: false
+  register: _systemd_version
```
Comment (paulfantom) [inline tasks/preflight.yml]: Use `node_exporter_systemd_version`

File: templates/node_exporter.service.j2 [hunk 2]
```diff
@@ -27,5 +27,18 @@ ExecStart=/usr/local/bin/node_exporter \
 SyslogIdentifier=node_exporter
 Restart=always
 
+PrivateTmp=yes
+ProtectHome=yes
+NoNewPrivileges=yes
+
+{% if prometheus_systemd_version.stdout | int >= 232 %}
```
Comment (paulfantom) [inline templates/node_exporter.service.j2]: Use `node_exporter_systemd_version.stdout`

File: tasks/preflight.yml (hunk 1 above)
Comment (paulfantom) [reply tasks/preflight.yml]: Also this task needs:
```
  tags:
    - skip_ansible_lint
```

Comment (ecksun) [reply tasks/preflight.yml]: done, note though that [install.yml](https://github.com/cloudalchemy/ansible-node-exporter/blob/master/tasks/install.yml#L30) uses the "_underscore" naming scheme

**Comment by ecksun** [comment]
I don't know if you get notifications for pushes, but I think I fixed your comments :)

**Pull request merged**